owsi-core/owsi-core-examples/wicket-showcase/wicket-showcase-webapp/src/main/java/fr/openwide/core/showcase/web/application/widgets/component/FileDownloadPanel.java

```diff
@@ -37,13 +38,20 @@ public FileDownloadPanel(String id) {
 		
 		add(
 				loadingPopup,
-				new AbstractFileDownloadAjaxLink("exportZip", loadingPopup, Model.of("export-showcase-"), Model.of(MediaType.APPLICATION_ZIP)) {
-					private static final long serialVersionUID = 1L;
-					@Override
-					protected File generateFile() throws ServiceException {
-						return generateTFile();
-					}
-				}
+				new FileDownloadAjaxLink("exportZip", loadingPopup,
+						new AbstractFileGenerationModel(
```

Comment:
I think you told me it's a bit too verbose for your taste.
Maybe you could add a constructor to `AbstractFileGenerationModel` that would accept the parameter of `ExportFileUtils.getFileNameMediaTypeModel` and would just call `ExportFileUtils.getFileNameMediaTypeModel` to build the model? Maybe even another one that would take a `String` and a `MediaType` as parameters (not models)?
Another solution, which might be cleaner: add a `SimpleFileDownloadAjaxLink` which would have a `generateFile` method and the constructors described above. Yes, it would be a step back to what we had before, but we'de still have `FileDownloadAjaxLink` for more complex needs.